crates/oas3/src/spec/schema.rs

```diff
@@ -628,4 +637,22 @@ mod tests {
         let schema = serde_yml::from_str::<ObjectSchema>(spec).unwrap();
         assert_eq!(schema.example, Some(serde_json::Value::Null));
     }
+
+    #[test]
+    fn discriminator_example_is_parsed_correctly() {
+        let spec = indoc::indoc! {"
+          oneOf:
+            - $ref: '#/components/schemas/Cat'
+            - $ref: '#/components/schemas/Dog'
+            - $ref: '#/components/schemas/Lizard'
+            - $ref: 'https://gigantic-server.com/schemas/Monster/schema.json'
+          discriminator:
+            propertyName: petType
+            mapping:
+            dog: '#/components/schemas/Dog'
+            monster: 'https://gigantic-server.com/schemas/Monster/schema.json'
```

Comment:
indentation incorrect and not caught by test due to lacking assertions on content